File: 백준/Silver/10866. 덱/덱.py
import collections
import sys
input = sys.stdin.readline

deque = collections.deque([])

orderCase = int(input())

for _ in range(orderCase) :
    order = input().rstrip()

    if order.startswith('push_back') : 
        deque.append(int(order[10:]))
    elif order.startswith('push_front') :
        deque.appendleft(int(order[10:]))
    elif order.startswith('pop_front') :
        try :
            print(deque.popleft())
        except :    
            print(-1)
    elif order.startswith('pop_back') :
        try :
            print(deque.pop())
        except :
            print(-1)
    elif order == 'size' : print(len(deque))
    elif order == 'empty' :
        if len(deque) == 0 : print(1)
        else : print(0)
    elif order == 'front' :
        if len(deque) != 0 : print(deque[0])
        else : print(-1)
    elif order == 'back' :
        if len(deque) != 0 : print(deque[-1])
        else : print(-1)
    

File: 백준/Silver/10866. 덱/덱.cc
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int N; cin >> N;
    deque<int> deque;

    string command;
    int x;
    for (int i = 0; i < N; i++) {
        cin >> command;
        
        if (command == "push_front") {
            cin >> x;
            deque.push_front(x);
        }
        else if (command == "push_back") { 
            cin >> x;
            deque.push_back(x);
        }
        else if (command == "pop_front") {
            if (deque.empty()) {
                cout << -1 << '\n'; 
                continue;
            }
            cout << deque.front() << '\n';
            deque.pop_front();
        }
        else if (command == "pop_back") {
            if (deque.empty()) {
                cout << -1 << '\n';
                continue;
            }
            cout << deque.back() << '\n';
            deque.pop_back();
        }
        else if (command == "size") cout << deque.size() << '\n';
        else if (command == "empty") {
            if (deque.empty()) cout << 1 << '\n';
            else cout << 0 << '\n';
        }
        else if (command == "front") {
            if (deque.empty()) cout << -1 << '\n';
            else cout << deque.front() << '\n'; 
        }        
        else {
            if (deque.empty()) cout << -1 << '\n';
            else cout << deque.back() << '\n';
        }
    }
}